src/Symfony/Component/AssetMapper/ImportMap/ImportMapManager.php

```diff
@@ -108,6 +108,34 @@ public function update(): array
         return $this->updateImportMapConfig(true, [], []);
     }
 
+    /**
+     * Downloads all missing downloaded packages.
+     *
+     * @return ImportMapEntry[] The downloaded packages
+     */
+    public function downloadMissingPackages(): array
+    {
+        $entries = $this->loadImportMapEntries();
+        $packagesToDownload = [];
+
+        foreach ($entries as $entry) {
+            if ($entry->isDownloaded && !$this->assetMapper->getAsset($entry->path)) {
+                $parts = self::parsePackageName($entry->importName);
+
+                $packagesToDownload[] = new PackageRequireOptions(
+                    $parts['package'],
+                    $parts['version'] ?? null,
```

Comment:
I think we should throw an exception if the `version` is null: we want to avoid updating the package on accident.

In another PR, I think we should start adding extra keys to each remote/vendor importmap entry: `version` & `package` so that we don't need to guess these from the URL.

```diff
    '@hotwired/turbo' => [
        'url' => 'https://cdn.jsdelivr.net/npm/@hotwired/turbo@7.3.0/+esm',
+       'package' => '@hotwired/turbo',
+       'version' => '7.3.0',
    ],
```

Perhaps even the `url` would "go away" in this case if the file was downloaded (so you'd have `download_to`, `package` and `version`).